pytest_mh/utils/firewall.py

```diff
@@ -289,81 +301,59 @@ def _next_priority(self) -> int:
         self._priority -= 1
         return priority
 
-    def add_direct_rule(
+    def add_nft_rule(
         self,
         chain: str,
         args: list[Any],
-        *,
-        table: str = "filter",
-        ip_family: Literal["ipv4", "ipv6", "all"] = "all",
-        priority: int | None = None,
-    ) -> int:
+    ) -> None:
         """
-        Add a new direct rule.
+        Add a new nft rule.
+
+        This method adds a nftables rule to the firewall. This rule is added without
+        the firewall knowing it but, if it is added to the firewall's table and chain,
+        it will be removed when the firewall is reloaded or restarted.
```

Comment:
The second sentence doesn't make much sense. Do you mean that if we add this rule to the firewall it'll be removed when the firewall is reloaded?

Reply:
The seconds sentence means the firewall is not aware of this rule because it didn't add it itself (as it was done before with firewall-cmd's direct rules.

Because the rule is added to a table and chain created by the firewall, they will be removed when the firewall is removed. I guess it simple erases the table with all the rules it has in it.